### PR TITLE
Fix magic time

### PR DESCRIPTION
**Pull request by Pragnya02**
In case of %%time with multi-lines in cell , eval the last line and display in output instead of suppressing it.

Refer Issue : https://github.com/ipython/ipython/issues/11707

### CODE REVIEW

**Comment by Carreau** [comment]
Looks good to me, 

I'll try to see if I can come up with test to make sure the behavior does not regress.

File: IPython/core/magics/execution.py [hunk 1]
```diff
@@ -1285,15 +1294,21 @@ def time(self,line='', cell=None, local_ns=None):
...
                 exec(code, glob, local_ns)
+                out=None
+                # multi-line %%time case
+                if expr_val and isinstance(expr_val, ast.Expr):                           
+                    out = eval(code_val, glob, local_ns)  
             except:
-                self.shell.showtraceback()
-                return
+                self.shell.showtraceback()               
+                return
```
Comment (gpotter2) [inline IPython/core/magics/execution.py]: If I'm not mistaken this only adds extra unneeded spaces.

Comment (Pragnya02) [reply IPython/core/magics/execution.py]: Did you mean showtraceback() ? That would be for the entire try block (from line 1301 to 1305)

Comment (gpotter2) [reply IPython/core/magics/execution.py]: What I meant was that your diff is full of spacing changes :smile:
B95A112B-B9A9-426A-BBE7-4396C0179C04

Comment (Pragnya02) [reply IPython/core/magics/execution.py]: Oops, thank you ! Fixed that now, should be fine.

**Comment by Carreau** [comment]
Thanks for your patience, 

I rebased and squashed a few commits to not have the whitespaces changes in history, plus added a test. 

Once test passes I'm happy to merge this. 

Thanks.

File: IPython/core/magics/execution.py [hunk 2]
```diff
@@ -1261,13 +1261,23 @@ def time(self,line='', cell=None, local_ns=None):
...
             mode = 'exec'
             source = '<timed exec>'
+            # multi-line %%time case
+            if len(expr_ast.body) > 1 :
+                expr_val=expr_ast.body[-1]
+                code_val = self.shell.compile(ast.Expression(expr_val.value)
+                                                , '<timed eval>'
+                                                , 'eval')
+                expr_ast=expr_ast.body[:-1]
+                expr_ast = ast.Module(expr_ast)
```
Comment (Carreau) [inline IPython/core/magics/execution.py]: That apparently is not compatible python 3.8+; I'd love to not break on a new Python.

Comment (Carreau) [reply IPython/core/magics/execution.py]: You need to do something similar to https://github.com/ipython/ipython/blob/master/IPython/core/interactiveshell.py#L110-L116

Comment (Carreau) [reply IPython/core/magics/execution.py]: I've done it.

**Comment by Carreau** [comment]
Thanks @Pragnya02 !

**Pull request merged**